manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml

```diff
@@ -115,3 +115,18 @@ spec:
         apiserver_request_count{group="extensions",version="v1beta1",resource!~"ingresses|",client!~"hyperkube/.*|cluster-policy-controller/.*"}
       labels:
         severity: warning
+  - name: apiserver-requests-in-flight
+    rules:
+      # We want to capture metrics for kube-apiserver and openshift-apiserver.
+      # namespace='default' indicates that the source is kube-apiserver.
+      # namespace='openshift-apiserver' indicates that the source is openshift-apiserver.
+      # After we aggregate, we add a new label 'apiserver' and map the values from existing 'namespace' label
+      # by chaining label_replace as follows:
+      #   'apiserver:kubernetes' if namespace = default
+      #   'apiserver:openshift' if namespace = openshift-apiserver
+    - record: cluster:apiserver_current_inflight_requests:sum
+      expr: |
+        label_replace(label_replace(sum(apiserver_current_inflight_requests{namespace=~"openshift-apiserver|default"}) by (requestKind,namespace), "apiserver", "openshift", "namespace", "openshift-apiserver"), "apiserver", "kubernetes", "namespace", "default")
```

Comment:
If this labeling is confusing/bothersome, I would prefer to fix that at discovery/collection time not just for telemetry. These metrics should be just as useful and comprehensible for someone querying them in-cluster as when we're sending them and if it's confusing for us then certainly it will be for users.

Reply:
I agree, so I have to change the corresponding `ServiceMonitor` object, correct? I will try this.

Reply:
Correct